src/aap_eda/api/serializers/activation.py

```diff
@@ -473,13 +471,168 @@ def create(self, validated_data):
                 "eda_system_vault_credential"
             ] = _create_system_eda_credential(
                 vault_data.password,
-                vault,
+                _get_vault_credential_type(),
                 validated_data.get("organization_id"),
             )
 
         return super().create(validated_data)
 
 
+class ActivationUpdateSerializer(serializers.ModelSerializer):
+    """Serializer for updating the Activation."""
+
+    class Meta:
+        model = models.Activation
+        fields = [
+            "name",
+            "description",
+            "decision_environment_id",
+            "rulebook_id",
+            "extra_var",
+            "organization_id",
+            "user",
+            "restart_policy",
+            "awx_token_id",
+            "log_level",
+            "eda_credentials",
+            "k8s_service_name",
+            "source_mappings",
+            "skip_audit_events",
+        ]
+
+    organization_id = serializers.IntegerField(
+        required=True,
+        allow_null=False,
+        validators=[validators.check_if_organization_exists],
+    )
```

Comment:
Please add customized error messages for `organization_id, rulebook_id` and `decision_environment_id`, similar as other serializers

Reply:
None of these fields are actually required for update.

Reply:
Are they editable? if so, we then need to validate them

Reply:
These fields are not required but cannot be null. I added customized error_messages for the null case.